### PR TITLE
Create test_optical_flow.py

### PR DESCRIPTION
**Pull request by kychennv**
add new case for testing optical flow with python, it should be passed when bug http://nvbugs/200520501 fixed

### CODE REVIEW

File: dali/test/python/test_optical_flow.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    except:
+       print("Unexpected error:", sys.exc_info())
+       
+copy_image(image_dir + "/frame_input.png", data_dir+"/0/0001.png")
+copy_image(image_dir + "/frame_reference.png", data_dir+"/0/0002.png")
+
+class OFPipeline(Pipeline):
+    def __init__(self, batch_size, num_threads, device_id):
+        super(OFPipeline, self).__init__(batch_size, num_threads, device_id, seed=16)
+        self.input = ops.SequenceReader(file_root=image_dir+"data", sequence_length=2)
```
Comment (klecki) [inline dali/test/python/test_optical_flow.py]:
```suggestion
        self.input = ops.SequenceReader(file_root=data_dir, sequence_length=2)
```

File: dali/test/python/test_optical_flow.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        self.of_op = ops.OpticalFlow(device="gpu", output_format=4)
+    def define_graph(self):
+        seq = self.input(name="Reader")
+        of = self.of_op(seq.gpu())
+        return of,seq.gpu()
+
+
+pipe = OFPipeline(batch_size=batch_size, num_threads=1, device_id=0)
```
Comment (klecki) [inline dali/test/python/test_optical_flow.py]: Can you convert it to use `nose`?
So this part will be a
```
def test_of:
	pipe = OFPipeline(batch_size=batch_size, num_threads=1, device_id=0)
	pipe.build()
	pipe_out = pipe.run()
	frames = pipe_out[0].as_cpu().as_array()
	print(frames.shape)
	np.savetxt("array.txt", frames.flatten(), fmt="%s")
	myarray=np.loadtxt(image_dir+'decoded_flow_vector.dat')
	assert (0.9 < np.mean(np.abs(frames[0][0].flatten()-myarray)))
```

And this test will be run with `nose`.
The copying of frames can be done as one of the `setup` or you can modify DALI_extra.

File: dali/test/python/test_optical_flow.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+def copy_image(source,target):
+    try:
+       shutil.copy(source, target)
+    except IOError as e:
+       print("Unable to copy file. %s" % e)
+    except:
+       print("Unexpected error:", sys.exc_info())
+       
+copy_image(image_dir + "/frame_input.png", data_dir+"/0/0001.png")
+copy_image(image_dir + "/frame_reference.png", data_dir+"/0/0002.png")
```
Comment (klecki) [inline dali/test/python/test_optical_flow.py]: You can modify DALI_extra to have:
`/db/optical_flow/two_frames/0/0001.png`, `optical_flow/two_frames/0/0002.png`
and skip all this mkdir and file copy.

Than use just `data_dir=dali_extra_path + "/db/optical_flow/two_frames"`.

File: dali/test/python/test_optical_flow.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import sys
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+from random import shuffle
+
+batch_size = 1
+sequence_length = 10
+dali_extra_path = os.environ['DALI_EXTRA_PATH']
+image_dir=dali_extra_path + "/db/optical_flow/slow_preset/"
```
Comment (JanuszL) [inline dali/test/python/test_optical_flow.py]:
```suggestion
image_dir= os.path.join(dali_extra_path, "/db/optical_flow/slow_preset/")
```

File: dali/test/python/test_optical_flow.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+data_dir=image_dir + "/data"
```
Comment (JanuszL) [inline dali/test/python/test_optical_flow.py]:
```suggestion
data_dir=os.path.join(image_dir, "/data")
```

File: dali/test/python/test_optical_flow.py (hunk 3 above)
Comment (kychennv) [reply dali/test/python/test_optical_flow.py]: do you mean i modify DALI_extra directly?

Comment (kychennv) [reply dali/test/python/test_optical_flow.py]: https://github.com/NVIDIA/DALI_extra/pull/13

File: dali/test/python/test_optical_flow.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import numpy as np
+import shutil
+import sys
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+from random import shuffle
+
+batch_size = 1
+sequence_length = 10
```
Comment (klecki) [inline dali/test/python/test_optical_flow.py]: This is unused.

Comment (szalpal) [inline dali/test/python/test_optical_flow.py]:
```suggestion
sequence_length = 2
```

File: dali/test/python/test_optical_flow.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+dali_extra_path = os.environ['DALI_EXTRA_PATH']
+image_dir= os.path.join(dali_extra_path, "/db/optical_flow/slow_preset/two_frames")
+
+
+class OFPipeline(Pipeline):
+    def __init__(self, batch_size, num_threads, device_id):
+        super(OFPipeline, self).__init__(batch_size, num_threads, device_id, seed=16)
+        self.input = ops.SequenceReader(file_root=image_dir, sequence_length=2)
```
Comment (szalpal) [inline dali/test/python/test_optical_flow.py]:
```suggestion
        self.input = ops.SequenceReader(file_root=image_dir, sequence_length=sequence_length)
```

**Comment by kychennv** [comment]
sorry that i could not fix the DCO error.  so i create another PR for it:
https://github.com/NVIDIA/DALI/pull/929,
please help to move to this PR.

File: dali/test/python/test_optical_flow.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        of = self.of_op(seq.gpu())
+        return of,seq.gpu()
+
+
+def test_of:
+	pipe = OFPipeline(batch_size=batch_size, num_threads=1, device_id=0)
+	pipe.build()
+	pipe_out = pipe.run()
+	frames = pipe_out[0].as_cpu().as_array()
+	print(frames.shape)
```
Comment (JanuszL) [inline dali/test/python/test_optical_flow.py]: Not needed.

**Comment by szalpal** [comment]
!build

**Comment by szalpal** [comment]
> @szalpal - do we have any L0 test for OF?
> IF not can you guide @kychennv hot to add one and execute this file from it?

Doesn't the qa scripts work that whatever is in dali/test/python gets ran?

**Comment by szalpal** [comment]
!build

**Comment by szalpal** [comment]
!build

File: qa/L0_python_self_test_TU102/test.sh [hunk 9]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash -e
+# used pip packages
+pip_packages="nose numpy opencv-python"
```
Comment (JanuszL) [inline qa/L0_python_self_test_TU102/test.sh]: Is opencv-python really used?

**Comment by szalpal** [comment]
!build

File: dali/test/python/test_optical_flow.py [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        super(OFPipeline, self).__init__(batch_size, num_threads, device_id, seed=16)
+        self.input = ops.SequenceReader(file_root=image_dir, sequence_length=sequence_length)
+        self.of_op = ops.OpticalFlow(device="gpu", output_format=4)
+    def define_graph(self):
+        seq = self.input(name="Reader")
+        of = self.of_op(seq.gpu())
+        return of,seq.gpu()
+
+
+def test_of:
```
Comment (JanuszL) [inline dali/test/python/test_optical_flow.py]:
```suggestion
def test_of():
```

**Comment by szalpal** [comment]
!build

**Pull request merged**